src/com/facebook/buck/features/apple/projectV2/FlagParser.java

```diff
@@ -195,6 +199,27 @@ void parseFlags(
                     targetNode.getBuildTarget(),
                     "swift_include_paths",
                     searchPathAttributes.swiftIncludeFlags()));
+
+        if (searchPathAttributes.vfsOverlay().isPresent()
+            && searchPathAttributes.vfsOverlayPath().isPresent()) {
+          VFSOverlay vfsOverlay = searchPathAttributes.vfsOverlay().get();
+          AbsPath vfsOverlayPath = searchPathAttributes.vfsOverlayPath().get();
+
+          if (Files.notExists(vfsOverlayPath.getParent().getPath())) {
+            Files.createDirectories(vfsOverlayPath.getParent().getPath());
+          }
+
+          if (Files.notExists(vfsOverlayPath.getPath())) {
+            try {
+              PrintWriter vfsOverlayFileWriter = new PrintWriter(vfsOverlayPath.getPath().toString());
+              vfsOverlayFileWriter.print(vfsOverlay.render());
+              vfsOverlayFileWriter.close();
+            } catch (FileNotFoundException e) {
+              e.printStackTrace();
+            }
+          }
+        }
```

Comment:
I see you are checking if the file exists to no-op here, but is it safe to do it?

What happens if you generate the project, then add a dependency to the module and generate the project again? Can you check if the VSFOverlay file will be updated?